api/client/build.go

```diff
@@ -104,12 +111,15 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 	case urlutil.IsGitURL(specifiedContext) && hasGit:
 		tempDir, relDockerfile, err = getContextFromGitURL(specifiedContext, *dockerfileName)
 	case urlutil.IsURL(specifiedContext):
-		tempDir, relDockerfile, err = getContextFromURL(cli.out, specifiedContext, *dockerfileName)
+		tempDir, relDockerfile, err = getContextFromURL(outBuf, specifiedContext, *dockerfileName)
 	default:
 		contextDir, relDockerfile, err = getContextFromLocalDir(specifiedContext, *dockerfileName)
 	}
 
 	if err != nil {
+		if *suppressOutput {
+			return fmt.Errorf("%s\nunable to prepare context: %s", outBuf, err)
```

Comment:
there are a couple of places where you add the output to the error, this is problematic because if there is specialized error processing where we expect just the error then we'll get extra text. For example, if you look at the daemon logs when the build fails you'll see the full build output + the error instead of just the error.

So, let's send the buffered output to the normal stream and then let the error be what it would be when -q isn't used. So in this case this line can be changed to something like:
```
cli.out.Write( outBuf )
```